pkg/controller/checknodehealth/controller.go

```diff
@@ -59,6 +61,11 @@ const (
 	ReasonPodStartupTimeout = "PodStartupTimeout"
 )
 
+var (
+	// ExpectedResults is the list of all expected health check results
+	ExpectedResults = []string{"PodStartup", "PodNetwork"}
```

Comment:
Now that this exists, I assume we only want to take into consideration the results from checkers listed here when determining whether the overall health is healthy/unhealthy/unknown.

Do `hasUnknownResult` and `hasUnhealthyResult` need to be updated to take this into account? If so, we should also update unit tests.

Reply:
I changed the name to `RequiredCheckResults` and added more comments to explain it.  

Regarding the `hasUnknownResult` and `hasUnhealthyResult`, I think you are right. I revise the logic to make it more clear. the logic is: if the the required checker is missing, then the result is Unknown by default.